polaris-react/src/components/ActionMenu/components/SecondaryAction/SecondaryAction.tsx

```diff
@@ -18,25 +18,39 @@ export function SecondaryAction({
   getOffsetWidth,
   ...rest
 }: SecondaryAction) {
-  const secondaryActionsRef = useRef<HTMLSpanElement>(null);
+  return (
+    <SecondaryActionWrapper
+      destructive={destructive}
+      getOffsetWidth={getOffsetWidth}
+    >
+      <Button onClick={onAction} {...rest}>
+        {children}
+      </Button>
+    </SecondaryActionWrapper>
+  );
+}
+
+export function SecondaryActionWrapper({
```

Comment:
Separated the wrapper to use the same style and width calculation logic for the custom buttons